pkg/apis/vsphere/v1/template_helpers.go

```diff
@@ -0,0 +1,73 @@
+package v1
+
+import (
+	"context"
+	"fmt"
+
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+	corev1 "go.anx.io/go-anxcloud/pkg/apis/core/v1"
+)
+
+var (
+	// ErrTemplateNotFound is returned when the named template was not found at a given location
+	ErrTemplateNotFound = fmt.Errorf("%w: named template was not found at specified location", api.ErrNotFound)
+)
+
+const (
+	// LatestTemplateBuild is used to find the template with the highest build number
+	LatestTemplateBuild = "latest"
+)
+
+// FindNamedTemplate retrieves a template by name and build at a specified location.
+// Empty and LatestTemplateBuild build identifier will yield the highest available build.
+// It returns ErrTemplateNotFound if no matching template was found.
+func FindNamedTemplate(ctx context.Context, a api.API, name, build string, location corev1.Location) (*Template, error) {
+	var match *Template
+	buildNo := -1
+	latest := build == "" || build == LatestTemplateBuild
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	var channel types.ObjectChannel
+
+	err := a.List(ctx, &Template{Type: TypeTemplate, Location: location}, api.ObjectChannel(&channel))
+	if err != nil {
+		return nil, fmt.Errorf("error listing templates: %w", err)
+	}
+
+	for res := range channel {
+		var template Template
+		err := res(&template)
+		if err != nil {
+			return nil, fmt.Errorf("error retrieving template: %w", err)
+		}
+
+		if template.Name != name {
+			continue
+		}
+
+		if latest {
+			currentTemplateBuildNo, err := template.BuildNumber()
+			if err != nil {
+				continue
```

Comment:
I think erroring out is what we want here, as we cannot know if we return the best possible template in this case

Reply:
Didn't see such build numbers yet and doubt they exist, but maybe .. can you at least log a warning? `ctx` should always have a logger attached (`logr.FromContextOrDiscard()`)